daemon/image_delete.go

```diff
@@ -232,7 +232,9 @@ func (daemon *Daemon) getContainerUsingImage(imageID image.ID) *container.Contai
 // optional tag or digest reference. If tag or digest is omitted, the default
 // tag is used. Returns the resolved image reference and an error.
 func (daemon *Daemon) removeImageRef(ref reference.Named) (reference.Named, error) {
-	ref = reference.WithDefaultTag(ref)
+	if reference.IsNameOnly(ref) {
+		ref = reference.EnsureTagged(ref)
+	}
```

Comment:
Sort of feel like this is a step backwards. Could we add a utility function `EnsureTaggedOrDigested` that does this?